content/shmem_atomic_swap_nbi.tex

```diff
@@ -0,0 +1,54 @@
+\color{Green}
+\apisummary{
+    This nonblocking atomic operation performs an atomic swap to a remote data
+    object.
+}
+
+\begin{apidefinition}
+
+\begin{C11synopsis}
+void @\FuncDecl{shmem\_atomic\_swap\_nbi}@(TYPE *fetch, TYPE *dest, TYPE value, int pe);
+void @\FuncDecl{shmem\_atomic\_swap\_nbi}@(shmem_ctx_t ctx, TYPE *fetch, TYPE *dest, TYPE value, int pe);
+\end{C11synopsis}
+where \TYPE{} is one of the extended \ac{AMO} types specified by
+Table~\ref{extamotypes}.
+
+\begin{Csynopsis}
+void @\FuncDecl{shmem\_\FuncParam{TYPENAME}\_atomic\_swap\_nbi}@(TYPE *fetch, TYPE *dest, TYPE value, int pe);
+void @\FuncDecl{shmem\_ctx\_\FuncParam{TYPENAME}\_atomic\_swap\_nbi}@(shmem_ctx_t ctx, TYPE *fetch, TYPE *dest, TYPE value, int pe);
+\end{Csynopsis}
+where \TYPE{} is one of the extended \ac{AMO} types and has a corresponding
+\TYPENAME{} specified by Table~\ref{extamotypes}.
+
+\begin{apiarguments}
+  \apiargument{IN}{ctx}{A context handle specifying the context on which to
+      perform the operation. When this argument is not provided, the
+      operation is performed on the default context.}
+  \apiargument{OUT}{fetch}{Local data object to be updated.}
+  \apiargument{OUT}{dest}{The remotely accessible data object to be
+    updated on the remote \ac{PE}.}
+  \apiargument{IN}{value}{The value to be atomically written to the remote
+    \ac{PE}.}
+  \apiargument{IN}{pe}{An integer that indicates the \ac{PE} number on which
+    \dest{} is to be updated.}
+\end{apiarguments}
+
+\apidescription{
+    The nonblocking \FUNC{shmem\_atomic\_swap\_nbi} routines perform an atomic
+    swap operation. This routine returns after posting the operation. The
```

Comment:
I wonder if it would be better to say "initiating" rather than "posting" since posting has a particular meaning in PCIe.